{{cookiecutter.service_name}}/service.py

```diff
@@ -445,7 +451,25 @@ def {{cookiecutter.workflow_id |replace("-", "_")  }}(conf, inputs, outputs): #
         )
         os.chdir(working_dir)
 
-        runner._namespace_name = "ws-" + inputs.get("workspace", {}).get("value", "default")
+        ## Identify the running namespace for the provided workspace ##
+
+        # Load kubeconfig
+        config.load_incluster_config()
+
+        # Create a CustomObjectsApi client instance
+        custom_api = client.CustomObjectsApi()
+
+        # Access the custom resource
+        workspace = custom_api.get_namespaced_custom_object(
+            group="core.telespazio-uk.io",
+            version="v1alpha1",
+            namespace="workspaces",
+            plural="workspaces",
+            name=inputs.get("workspace", {}).get("value", "default"),
+        )
+
+        #runner._namespace_name = "ws-" + inputs.get("workspace", {}).get("value", "default")
+        runner._namespace_name = workspace["spec"]["namespace"]
```

Comment:
I think that error handling and the "default" case needs a bit of a rethink:
- I don't think we want to handle the case when no workspace is given and it tries to use "default" any more?
- get_namespaced_custom_object could error if the given workspace doesn't exist - We want to error there but rethrowing with a nice log statement might be good.

Reply:
Good shout, I have cleaned this up and removed the "default" value. It should now error if the workspace name is not defined or if the workspace CRD does not exist. The error is then logged appropriately.